presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/MultiKeyValuePairStateSerializer.java

```diff
@@ -43,14 +47,13 @@ public void serialize(MultiKeyValuePairsState state, BlockBuilder out)
             out.appendNull();
         }
         else {
-            //  FIXME: This assumes MapType allows duplicate keys when writing object out
-            mapType.writeObject(out, state.get().serialize());
+            serializedType.writeObject(out, state.get().serialize());
```

Comment:
It looks like it will be a simple change to make serialize take a `BlockBuilder`. Let's do it.